src/scripts/page/form/demo.js

```diff
@@ -0,0 +1,192 @@
+import {BindingEngine, inject} from 'aurelia-framework';
+
+@inject(BindingEngine)
+export class Demo {
+
+
+  constructor(bindingEngine) {
+    this.petOwnerForm = petOwnerForm(bindingEngine);
+    this.loginForm = loginForm();
+    this.productForm = productForm();
+  }
+
+}
+
+function productForm() {
+  let model = {
+    name: 'Bread',
+    tags: [
+      {
+        name: 'Basics',
+        color: '#FF3333'
+      },
+      {
+        name: 'Brown',
+        color: '#992233'
+      }
+    ]
+  };
+
+  let removeTag = (tag, index, $event) => {
+    console.log(tag, index, $event);
+    model.tags.splice(index, 1);
+  };
+
+  let addTag = (tag, index, $event) => {
+    console.log('addTag', tag, index, $event);
+    model.tags.push({
+      name: '',
+      color: '#ffffff'
+    });
+  };
+
+  let tagSchema = [
+    {
+      key: 'name',
+      type: 'string',
+      attributes: {
+        placeholder: 'tag name'
+      }
+    },
+    {
+      key: 'color',
+      type: 'color'
+    },
+    {
+      type: 'buttons',
+      actions: [{
+        label: 'remove',
+        attributes: {
+          class: 'btn btn-warning',
+        },
+        action: removeTag
+      }]
+    }
+  ];
+
+  let schema = [
+    {
+      key: 'name',
+      type: 'string'
+    },
+    {
+      key: 'tags',
+      type: 'collection',
+      schema: tagSchema
+    },
+    {
+      type: 'buttons',
+      actions: [{
+        label: 'add tag',
+        action: addTag,
+        attributes: {
+          class: 'btn btn-primary'
+        }
+      }]
+    }
+  ];
+
+  return {schema, model};
+}
+
+function petOwnerForm(bindingEngine) {
+
+  let petsModel = ['cat', 'dog'];
+
+  let petsOptions = [];
+
+  let calculatePetsOptions = () => {
+    petsOptions.length = 0; /* empty the array */
```

Comment:
It is an array. element.options takes an array to iterate and generate the visible options.